pkg/prometheus/promcfg.go

```diff
@@ -2257,29 +2256,39 @@ func (cg *ConfigGenerator) appendEvaluationInterval(slice yaml.MapSlice, evaluat
 	return append(slice, yaml.MapItem{Key: "evaluation_interval", Value: evaluationInterval})
 }
 
+func (cg *ConfigGenerator) appendGlobalLimits(slice yaml.MapSlice, version string, limitKey string, limit *uint64, enforcedLimit *uint64) yaml.MapSlice {
+	if limit == nil && enforcedLimit != nil && *enforcedLimit > 0 {
```

Comment:
(suggestion) and you can get rid of the next block IIUC
```suggestion
	if ptr.Deref(limit, 0) == 0 && ptr.Deref(enforcedLimit, 0) > 0 {
```